django/db/models/sql/compiler.py

```diff
@@ -387,18 +387,19 @@ def _order_by_pairs(self):
                     True,
                 )
                 continue
-            if col in self.query.annotations:
-                # References to an expression which is masked out of the SELECT
-                # clause.
-                if self.query.combinator and self.select:
+
+            ref, *transforms = col.split(LOOKUP_SEP)
+            if expr := self.query.annotations.get(ref):
+                if transforms:
+                    for name in transforms:
+                        expr = self.query.try_transform(expr, name)
+                elif self.query.combinator and self.select:
```

Comment:
Maybe we should error out if trying to order with a transform from a combined query as this is not supported.

Reply:
Agreed.